File: cells/fast_math.py
import math
import numpy as np

# Precompute LUT with 0.1° resolution (3600 points)
_SIN_LUT = [math.sin(math.radians(angle/10)) for angle in range(0, 3600)]
_COS_LUT = [math.cos(math.radians(angle/10)) for angle in range(0, 3600)]

def process(input_data):
    """Returns sin/cos with 0.1° precision using precomputed LUTs"""
    angle_deg = input_data["angle"] % 360.0  # Normalize to [0,360)
    idx = int(round(angle_deg * 10)) % 3600  # Convert to LUT index
    
    return {
        "sin": _SIN_LUT[idx],
        "cos": _COS_LUT[idx],
        "precision_deg": 0.1,
        "method": "LUT"
    }

File: cells/fast_math.js
// Precomputed LUTs for 0.1° resolution
const _sin_lut = Array.from({length: 3600}, (_, i) => 
    Math.round(Math.sin(Math.PI * i / 1800) * 1000)
);

export const config = {
    id: "fast_math",
    name: "Fast Trigonometry",
    category: "utils",
    description: "High-speed sin/cos calculations",
    inputs: [
        { id: "angle", label: "Angle (degrees)", type: "number", value: 45 }
    ]
};

export function process(input_data) {
    const angle = input_data["angle"] % 360;
    const idx = Math.floor(angle * 10);
    return {
        sin: _sin_lut[idx] / 1000,
        cos: _sin_lut[(idx + 900) % 3600] / 1000
    };
}
